From fc9a244e5ba9164d00b1ee714608174806ff39dc Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 21 Dec 2019 07:04:10 +0200
Subject: [PATCH] [SW-8362] hl: lds: use macros for text base and size

This way moving around binary e.g. scratchpad sram can be done by
only changing the macro of {PP,PRE}BOOT_TEXT_BASE

Change-Id: Ia43022f40b5c2a43f956350581c8fbdb6da4b215
---

diff --git a/gaudi/common.h b/gaudi/common.h
index b9b1a05..4ef9711 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -5,7 +5,10 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#define PREBOOT_TEXT_BASE		0x7FF0004000ull
+#include <preboot.h>
+
+#define PPBOOT_TEXT_BASE		0x7FF0000000
+#define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
 #define mmDMA_W_PLL_BASE		0x487000
 #define mmIF_W_PLL_BASE			0x488000
diff --git a/gaudi/ppboot.lds.S b/gaudi/ppboot.lds.S
index 8226163..68d58f1 100644
--- a/gaudi/ppboot.lds.S
+++ b/gaudi/ppboot.lds.S
@@ -5,7 +5,9 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-MEMORY { .sram : ORIGIN = 0x7FF0000000, LENGTH = 0x10000 }
+#include "common.h"
+
+MEMORY { .sram : ORIGIN = PPBOOT_TEXT_BASE, LENGTH = PPBOOT_SIZE }
 
 OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
 OUTPUT_ARCH(aarch64)
diff --git a/gaudi/preboot.lds.S b/gaudi/preboot.lds.S
index 61e3d10..01650fd 100644
--- a/gaudi/preboot.lds.S
+++ b/gaudi/preboot.lds.S
@@ -5,7 +5,9 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-MEMORY { .sram : ORIGIN = 0x7FF0004000, LENGTH = 0x10000 }
+#include "common.h"
+
+MEMORY { .sram : ORIGIN = PREBOOT_TEXT_BASE, LENGTH = PREBOOT_SIZE }
 
 OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
 OUTPUT_ARCH(aarch64)
diff --git a/goya/common.h b/goya/common.h
index 334d52c..308d933 100644
--- a/goya/common.h
+++ b/goya/common.h
@@ -5,7 +5,10 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#define PREBOOT_TEXT_BASE	0x7FF0004000ull
+#include <preboot.h>
+
+#define PPBOOT_TEXT_BASE	0x7FF0000000
+#define PREBOOT_TEXT_BASE	(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
 #define mmCPU_PLL_BASE		0x4A2000
 #define mmIC_PLL_BASE		0x4A3000
diff --git a/goya/ppboot.lds.S b/goya/ppboot.lds.S
index ecaa722..68d58f1 100644
--- a/goya/ppboot.lds.S
+++ b/goya/ppboot.lds.S
@@ -5,7 +5,9 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-MEMORY { .sram : ORIGIN = 0x7ff0000000, LENGTH = 0x10000 }
+#include "common.h"
+
+MEMORY { .sram : ORIGIN = PPBOOT_TEXT_BASE, LENGTH = PPBOOT_SIZE }
 
 OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
 OUTPUT_ARCH(aarch64)
diff --git a/goya/preboot.lds.S b/goya/preboot.lds.S
index 8e43cc0..01650fd 100644
--- a/goya/preboot.lds.S
+++ b/goya/preboot.lds.S
@@ -5,7 +5,9 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-MEMORY { .sram : ORIGIN = 0x7ff0000000, LENGTH = 0x10000 }
+#include "common.h"
+
+MEMORY { .sram : ORIGIN = PREBOOT_TEXT_BASE, LENGTH = PREBOOT_SIZE }
 
 OUTPUT_FORMAT("elf64-littleaarch64", "elf64-littleaarch64", "elf64-littleaarch64")
 OUTPUT_ARCH(aarch64)
diff --git a/goya2_arc/common.h b/goya2_arc/common.h
index 334d52c..9ebee60 100644
--- a/goya2_arc/common.h
+++ b/goya2_arc/common.h
@@ -5,7 +5,10 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#define PREBOOT_TEXT_BASE	0x7FF0004000ull
+#include <preboot.h>
+
+#define PPBOOT_TEXT_BASE	0xf0000000
+#define PREBOOT_TEXT_BASE	(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
 #define mmCPU_PLL_BASE		0x4A2000
 #define mmIC_PLL_BASE		0x4A3000
diff --git a/goya2_arc/ppboot.lds.S b/goya2_arc/ppboot.lds.S
index d95ae31..9115076 100644
--- a/goya2_arc/ppboot.lds.S
+++ b/goya2_arc/ppboot.lds.S
@@ -5,8 +5,10 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
+#include "common.h"
+
 #ifdef __arc__
-MEMORY { .sram : ORIGIN = 0xf0000000, LENGTH = 0x10000 }
+MEMORY { .sram : ORIGIN = PPBOOT_TEXT_BASE, LENGTH = PPBOOT_SIZE }
 OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
 OUTPUT_ARCH(arc)
 #endif
diff --git a/goya2_arc/preboot.lds.S b/goya2_arc/preboot.lds.S
index 34a85b2..03c9bf9 100644
--- a/goya2_arc/preboot.lds.S
+++ b/goya2_arc/preboot.lds.S
@@ -5,8 +5,10 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
+#include "common.h"
+
 #ifdef __arc__
-MEMORY { .sram : ORIGIN = 0xf0000000, LENGTH = 0x10000 }
+MEMORY { .sram : ORIGIN = PREBOOT_TEXT_BASE, LENGTH = PREBOOT_SIZE }
 OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
 OUTPUT_ARCH(arc)
 #endif
diff --git a/include/preboot.h b/include/preboot.h
index fbd9a68..ce48b77 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -25,7 +25,8 @@
 #define SPL_TEXT_BASE				0x7FF0080000ull
 #endif
 
-#define PREBOOT_SIZE		0x4000		/* 16KB */
+#define PPBOOT_SIZE		0x4000		/* 16KB */
+#define PREBOOT_SIZE		0xC000		/* 48KB */
 
 #define SPL_SIZE		0x10000		/* 64KB */
 #define HL_UART_BAUDRATE	115200
@@ -37,6 +38,8 @@
 #define HL_SPI_PREBOOT_OFF	0x1A0000
 #define HL_SPI_PRIMARY_BASE	0
 #define HL_SPI_SECONDARY_BASE	0x800000
+
+#ifndef __ASSEMBLY__
 
 typedef unsigned char u8;
 typedef unsigned int u32;
@@ -87,4 +90,6 @@
 void preboot_putc_virt_uart(const char c);
 void preboot_sync_btl(void);
 
+#endif /* __ASSEMBLY__ */
+
 #endif /* COMMON_PREBOOT_H_ */
